services/web/server/src/simcore_service_webserver/diagnostics_monitoring.py

```diff
@@ -131,6 +137,11 @@ async def _middleware_handler(request: web.Request, handler: Callable):
 def setup_monitoring(app: web.Application):
     # app-scope registry
     app[kCOLLECTOR_REGISTRY] = reg = CollectorRegistry(auto_describe=True)
+    # creating the pool at setup
+
+    app[  # pylint: disable=consider-using-with
```

Comment:
this threadpool is not closed properly. Actually pylint is warning you

To init, you should use: https://docs.aiohttp.org/en/stable/web_advanced.html#aiohttp-web-cleanup-ctx

```python

async def setup_thread_pool(app: web.Application):
   with ThreadPOolExectuor(...) as pool:
        app[KTHREAD_POOL_EXECUTOR] = pool
        yield


def setup_monitoring(app):
   # ...
   app.cleanup_ctx.append(setup_thread_pool)
  # ...

```

Reply:
This is a good one! Thanks.